src/Workspaces/CSharp/Portable/Extensions/SemanticModelExtensions.cs

```diff
@@ -346,6 +372,25 @@ private static ImmutableArray<ParameterName> GenerateNames(IList<string> reserve
             return GenerateNames(reservedNames, isFixed, parameterNames);
         }
 
+        public static ImmutableArray<ParameterName> GenerateParameterNames(
```

Comment:
two GenerateParamaeterName could be merged? GenerateParameterName<T>(.... Func<T, bool> canBeFixed)
only place 2 code is different is a => a.NameColon and a.NameEqual so, it could be changed to a => canBeFixed(a) ?

Reply:
Good idea.  I opened this issue for the refactoring: https://github.com/dotnet/roslyn/issues/33726